mypy/checker.py

```diff
@@ -1833,6 +1833,11 @@ def check_assignment(self, lvalue: Lvalue, rvalue: Expression, infer_lvalue_type
                         # Try to infer a partial type. No need to check the return value, as
                         # an error will be reported elsewhere.
                         self.infer_partial_type(lvalue_type.var, lvalue, rvalue_type)
+                    # Handle PartialType's super type checking here, after it's resolved.
+                    if (isinstance(lvalue, RefExpr) and
+                            self.check_compatibility_all_supers(lvalue, lvalue_type, rvalue)):
```

Comment:
The exception you added above removes the check for all kinds of partial types, while this only re-adds the check for partial `None` types. I don't say it is necessarily wrong, but I would at least add test cases with partial list and dict types (e.g. with an empty list in subclass that is subsequently gets an `.append()`).

Reply:
Yes, you are right. I will change the code, actually.

Reply:
Are you going to add the tests with an empty list in the subclass?

Reply:
I don't think I should add them anymore, because now I make the exception for only `None` partial types.

Reply:
But there may be a similar crash (or a regression later), I think it is better to have one in any case.

Reply:
I tried this on both master and this branch, it didn't give an error:

```py
from typing import List

class A:
    x = ['a', 'b']

class B(A):
    x = []
    x.append(2)
```

I think it should give an error, but I could not be sure. What do you think? @ilevkivskyi

Reply:
Yes, this should be an error. It looks like another consequence of https://github.com/python/mypy/issues/4547. Let's then merge this issue as is, and consider adding this test when https://github.com/python/mypy/issues/4547 is fixed. Could you please add a comment in that issue?